src/db/tables/subscribers.js

```diff
@@ -148,6 +150,49 @@ async function updateFxAData (subscriber, fxaAccessToken, fxaRefreshToken, fxaPr
 }
 /* c8 ignore stop */
 
+/**
+ * Update fxa tokens for subscriber
+ *
+ * @param {any} subscriber knex object in DB
+ * @param {string | null} fxaAccessToken from Firefox Account Oauth
+ * @param {string | null} fxaRefreshToken from Firefox Account Oauth
+ * @param {number} sessionExpiredAt from Firefox Account Oauth
+ * @returns {Promise<any>} updated subscriber knex object in DB
+ */
+// Not covered by tests; mostly side-effects. See test-coverage.md#mock-heavy
+/* c8 ignore start */
+async function updateFxATokens (subscriber, fxaAccessToken, fxaRefreshToken, sessionExpiredAt) {
+  const updated = await knex('subscribers')
+    .where('id', '=', subscriber.id)
+    .update({
+      fxa_access_token: fxaAccessToken,
+      fxa_refresh_token: fxaRefreshToken,
+      fxa_session_expiry: new Date(sessionExpiredAt),
+      // @ts-ignore knex.fn.now() results in it being set to a date,
+      // even if it's not typed as a JS date object:
+      updated_at: knex.fn.now(),
+    })
+    .returning('*')
+  const updatedSubscriber = Array.isArray(updated) ? updated[0] : null
```

Comment:
Is it possible that `updated` can be an empty array?
Although i think accessing `[0]` on an empty array just returns `undefined` and doesn't throw an error.

Reply:
updated it with a length check